keps/973-workload-priority/README.md

```diff
@@ -0,0 +1,225 @@
+
+Implement `WorkloadPriorityClass`.
+`Workload` can utilize `WorkloadPriorityClass`.
+
+### Non-Goals
+
+Using existing k8s `PriorityClass` for `Workload` is not recommended.
+`WorkloadPriorityClass` doesn't implement all the features of the k8s `PriorityClass` because some fields on the k8s `PriorityClass` are not relevant to Kueue.
+
+## Proposal
+
+In this proposal, a `WorkloadPriorityClass` will be created.
+The `Workload` will be able to utilize this `WorkloadPriorityClass`.
+`WorkloadPriorityClass` is independent from pod's priority.
+The priority value will be part of the workload spec and be mutable.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
+
+Kueue issue [973](https://github.com/kubernetes-sigs/kueue/issues/973) provides details on the initial feature implementation.
+
+#### Story 1
+
+In an organization, admins want to set a lower priority for development workloads and a higher priority for production workloads.
+In such cases, they create two `WorkloadPriorityClass` and apply each one to the respective workloads.
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha1
+kind: WorkloadPriorityClass
+metadata:
+  name: dev-workload-class
+value: 100
+description: "Dev's priority class"
+---
+apiVersion: kueue.x-k8s.io/v1alpha1
+kind: WorkloadPriorityClass
+metadata:
+  name: prod-workload-class
+value: 10000
+description: "Prod's priority class"
+---
+# dev-workload-a.yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: Workload
+metadata:
+  name: dev-job-a
+spec:
+  priorityClassName: dev-workload-class
+  queueName: team-a-queue
+  podSets:
+  - count: 3
+    name: main
+    template:
+      spec:
+        containers:
+        - image: gcr.io/k8s-staging-perf-tests/sleep:latest
+---        
+# dev-workload-b.yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: Workload
+metadata:
+  name: dev-job-b
+spec:
+  priorityClassName: dev-workload-class
+  queueName: team-b-queue
+  podSets:
+  - count: 3
+    name: main
+    template:
+      spec:
+        containers:
+        - image: gcr.io/k8s-staging-perf-tests/sleep:latest
+# prod-workload-a.yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: Workload
+metadata:
+  name: prod-job-b
+spec:
+  priorityClassName: prod-workload-class
+  queueName: team-b-queue
+  podSets:
+  - count: 3
+    name: main
+    template:
+      spec:
+        containers:
+        - image: gcr.io/k8s-staging-perf-tests/sleep:latest        
+```
+
+
+
+### Risks and Mitigations
+
+It's possible that the pod's priority conflicts with the workload's priority.
+For example, a high-priority job with low-priority pods may never run to completion because it may always be preempted.
```

Comment:
```suggestion
For example, a high-priority job with low-priority pods may never run to completion because it may always be preempted by kube-scheduler.
```